packages/medusa/src/api/routes/admin/orders/create-order.ts

```diff
@@ -0,0 +1,204 @@
+ *             description: The Shipping Method used for shipping the order.
+ *             type: object
+ *             required:
+ *               - provider_id
+ *               - profile_id
+ *               - price
+ *             properties:
+ *               provider_id:
+ *                 type: string
+ *                 description: The id of the shipping provider.
+ *               profile_id:
+ *                 type: string
+ *                 description: The id of the shipping profile.
+ *               price:
+ *                 type: integer
+ *                 description: The price of the shipping.
+ *               data:
+ *                 type: object
+ *                 description: Data relevant to the specific shipping method.
+ *               items:
+ *                 type: array
+ *                 description: Items to ship
+ *           no_notification:
+ *             description: A flag to indicate if no notifications should be emitted related to the updated order.
+ *             type: boolean
+ * tags:
+ *   - Order
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             order:
+ *               $ref: "#/components/schemas/order"
+ */
+
+export default async (req, res) => {
+  const validated = await validator(AdminPostOrderReq, req.body)
+
+  const orderService = req.scope.resolve("orderService")
+  let order = await orderService.create(validated)
+  order = await orderService.decorate(order, [], ["region"])
+
+  res.status(200).json({ order })
+}
+
+export class AdminPostOrderReq {
+  @IsString()
+  @IsOptional()
+  status?: string
+
+  @IsEmail()
+  @IsNotEmpty()
+  email: string
+
+  @IsNotEmpty()
+  @ValidateNested()
+  @Type(() => AddressPayload)
+  billing_address: AddressPayload
+
+  @IsNotEmpty()
+  @ValidateNested()
+  @Type(() => AddressPayload)
+  shipping_address: AddressPayload
+
+  @IsArray()
+  @IsNotEmpty()
+  items: object[]
+
+  @IsString()
+  @IsNotEmpty()
+  region: string
+
+  @IsArray()
+  @IsOptional()
+  discounts?: object[]
+
+  @IsString()
+  @IsNotEmpty()
+  customer_id: string
+
+  @IsNotEmpty()
+  @ValidateNested()
+  @Type(() => PaymentMethod)
+  payment_method: PaymentMethod
+
+  @IsNotEmpty()
+  @ValidateNested({ each: true })
+  @Type(() => ShippingMethod)
+  shipping_method?: ShippingMethod[]
+
+  @IsBoolean()
+  @IsOptional()
+  @Transform(({ value }) => value && value.toString() === "true")
+  no_notification?: boolean
+}
+
+class PaymentMethod {
+  @IsString()
+  @IsNotEmpty()
+  provider_id: string
+
+  @IsObject()
+  @IsOptional()
+  data?: object
+}
+
+class ShippingMethod {
+  @IsString()
+  @IsNotEmpty()
+  provider_id: string
+
+  @IsString()
+  @IsNotEmpty()
+  profile_id: string
+
+  @IsInt()
+  @IsNotEmpty()
+  @Type(() => Number)
+  price: number
+
+  @IsObject()
+  @IsOptional()
+  data?: object
+
+  @IsArray()
+  @IsOptional()
+  items?: object[]
```

Comment:
Question: Shouldn't this be a specific type, or is it intentionally just `object`?